rules/utils/get-number-literal-type.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+/**
+ * Return the type of the number in the node
+ * @param {Node} node - The node to check. Its type must be "Literal", and its value must be a number.
+ * @returns {undefined|"bigint"|"hexadecimal"|"binary"|"octal"|"number"}
+ */
+function getNumberLiteralType(node) {
+	if (!node?.type === 'Literal' || !['number', 'bigint'].includes(typeof node.value)) {
```

Comment:
Did you mean `node?.type !== 'Literal'`? But why `?.` Node.type  always exits.

Reply:
I did mean `node?.type !== 'Literal'`, you're correct.
This is a function and everything can be passed here, thus node might not be a Node object. Should we consider that `node` is always an instance of `Node`? Should I check if `node instanceof Node`? Because it's true that currently it is not consistent at all.
Anyway the optional chaining syntax should be removed, because it is the cause of all those failing CI (which I'm currently fixing)

Reply:
If you don't pass something that's not a Node, I don't think you need add this test. If someone need check  it, he can add it when he use this.  Normally we do `!node ||`